functions/Makefile

```diff
@@ -0,0 +1,38 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+SHELL=/bin/bash
+TAG := latest
+
+.DEFAULT_GOAL := help
+.PHONY: help
+help: ## Print this help
+	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
+
+.PHONY: test build build-dev push
+
+test: ## Run unit tests for all functions
+	cd go && $(MAKE) test
+	cd ts && $(MAKE) test
+
+build: ## Build all function images. Variable 'TAG' is used to specify tag. 'latest' will be used if not set.
+	cd go && $(MAKE) TAG=$(TAG) build
```

Comment:
IMO we should at least have a build target that doesn't invoke linters, otherwise it makes our CD pipeline setup a little more complicated. The linters require the go toolchain while building images requires the docker toolchain. When using GCB, if we can have seperate make target for those, we can simply use GCB's images (e.g. gcr.io/cloud-builder/go and gcr.io/cloud-builder/docker); otherwise, we may need to maintain our own build images that contains both docker toolchain and golang toolchain.

My suggestion would be at the top level Makefile having the `build` only build the images, while the `lint` or 'verify' target invokes the linters. We can have a `all` target that does lint first then build. In each child level, we also separate these 2.

Reply:
But building TS images requires a node environment.